io-engine/src/core/bdev.rs

```diff
@@ -167,12 +174,48 @@ where
                 bytes_written: stat.bytes_written,
                 num_unmap_ops: stat.num_unmap_ops,
                 bytes_unmapped: stat.bytes_unmapped,
+                read_latency_ticks: stat.read_latency_ticks,
+                max_read_latency_ticks: stat.max_read_latency_ticks,
+                min_read_latency_ticks: stat.min_read_latency_ticks,
+                write_latency_ticks: stat.write_latency_ticks,
+                max_write_latency_ticks: stat.max_write_latency_ticks,
+                min_write_latency_ticks: stat.min_write_latency_ticks,
+                max_unmap_latency_ticks: stat.max_unmap_latency_ticks,
+                min_unmap_latency_ticks: stat.min_unmap_latency_ticks,
+                unmap_latency_ticks: stat.unmap_latency_ticks,
+                tick_rate: self.get_tick_rate(),
             }),
             Err(err) => Err(CoreError::DeviceStatisticsFailed {
                 source: err,
             }),
         }
     }
+
+    /* pub async fn reset_stats_old(&self) -> Result<(), CoreError> {
+        match self.inner.stats_async().await {
+            Ok(mut stat) => {
+                info!("Got stats for bdev, {}", self.name());
+                let _ = unsafe {
+                    spdk_bdev_reset_io_stat(&mut stat, BdevResetMode::SpdkBdevResetStatAll as u32)
+                };
+                info!("Reset completed for bdev {}", self.name());
+                return Ok(())
+            }
+            Err(err) => {return Err(CoreError::DeviceStatisticsFailed {
+                source: err,
+            })},
+        }
+    } */
+
+    pub async fn reset_bdev_io_stats(&self) -> Result<(), CoreError> {
```

Comment:
pub async fn reset_bdev_io_stats(&self) -> Result<(), CoreError> {
    if let Err(err) = self.inner.stats_reset_async().await {
        return Err(CoreError::DeviceStatisticsFailed {
            source: err,
        });
    }
    Ok(())
}
Better to have a default return from the function as shown here.